library/library.cpp

```diff
@@ -98,6 +100,26 @@ namespace bzapi
         return std::make_pair(addr, port);
     }
 
+    static bool
+    check_for_empty_addresses(const std::string& esr_address, const std::string& url)
```

Comment:
Could you please rename this to something a bit more generic such as validate_addresses. The current name leaks the implementation a bit too much for my liking.